jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeTable.kt

```diff
@@ -207,7 +207,8 @@ class S3TreeTable(
         return (path.lastPathComponent as DefaultMutableTreeNode).userObject as? S3TreeNode
     }
 
-    fun getRootNode(): S3TreeDirectoryNode = (tableModel.root as DefaultMutableTreeNode).userObject as S3TreeDirectoryNode
+    // This is nullable because tableModel.root is nullable. It is null sometimes, like when we invalidate the tree on a refresh
+    fun getRootNode(): S3TreeDirectoryNode? = (tableModel.root as? DefaultMutableTreeNode)?.userObject as? S3TreeDirectoryNode
```

Comment:
We never actually allow root node to be null / replaced, Why not just keep a reference to it in this class instead of going through the model?

Reply:
That makes a lot of sense, changed